src/components/sections/EditorSection.tsx

```diff
@@ -0,0 +1,45 @@
+import { useAlert, useDataQuery } from '@dhis2/app-runtime'
+import React, { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import Editor from './Editor'
+
+const EditorSection = ({ query }) => {
+    const { key, namespace, store } = useParams()
+    const { show: showError } = useAlert('An error fetching this data', {
+        critical: true,
+    })
+
+    const { data, loading, refetch } = useDataQuery(query, {
+        variables: {
+            key,
+            namespace,
+        },
+        onError: () => showError(),
+    })
+
+    const [value, setValue] = useState(
+        JSON.stringify(data?.results, null, 4) || ''
+    )
+
+    useEffect(() => {
+        setValue(JSON.stringify(data?.results, null, 4))
+    }, [data])
```

Comment:
Why is this in state and updated using `useEffect`, I would expect you could just do `useMemo` for this? It's calculated value based on data?

Reply:
The latter as used [here](https://github.com/dhis2/datastore-app/blob/df09feec85a9b81b1d6bcf80d1d91c95b7b55bda/src/components/sections/EditSection.tsx#L40)